rasterio/env.py

```diff
@@ -327,38 +327,146 @@ def __lt__(self, other):
         return (self.major, self.minor) < tuple(other.major, other.minor)
 
     def __repr__(self):
-        return "GDALVersion(major={major}, minor={})".format(self)
+        return "GDALVersion(major={0}, minor={1})".format(self.major, self.minor)
+
+    def __str__(self):
+        return "{0}.{1}".format(self.major, self.minor)
 
     @classmethod
-    def _normalize(cls, input):
-        """Normalize input tuple or string to GDALVersion"""
+    def parse(cls, input):
+        """
+        Parses input tuple or string to GDALVersion. If input is a GDALVersion
+        instance, it is returned.
 
+        Parameters
+        ----------
+        input: tuple of (major, minor), string, or instance of GDALVersion
+
+        Returns
+        -------
+        GDALVersion instance
+        """
+
+        if isinstance(input, cls):
+            return input
         if isinstance(input, tuple):
             return cls(*input)
         elif isinstance(input, string_types):
-            return cls.from_string(input)
-        elif not isinstance(input, cls):
-            raise TypeError("Can only compare GDALVersion to same class, "
-                             "tuple, or string")
-        return input
-
-    @classmethod
-    def from_string(cls, s):
-        """Extract major and minor version components.
-        alpha, beta, rc suffixes ignored"""
-        match = re.search('^\d+\.\d+', s)
-        if not match:
-            raise ValueError("value does not appear to be a valid GDAL version "
-                             "number: {}".format(s))
-        major, minor = (int(c) for c in match.group().split('.'))
-        return cls(major=major, minor=minor)
+            # Extract major and minor version components.
+            # alpha, beta, rc suffixes ignored
+            match = re.search('^\d+\.\d+', input)
+            if not match:
+                raise ValueError(
+                    "value does not appear to be a valid GDAL version "
+                    "number: {}".format(input))
+            major, minor = (int(c) for c in match.group().split('.'))
+            return cls(major=major, minor=minor)
+
+        raise TypeError("GDALVersion can only be parsed from a string or tuple")
 
     @classmethod
     def runtime(cls):
         """Return GDALVersion of current GDAL runtime"""
         from rasterio._base import gdal_version  # to avoid circular import
-        return cls.from_string(gdal_version())
+        return cls.parse(gdal_version())
 
     def at_least(self, other):
-        other = self.__class__._normalize(other)
-        return self >= other
\ No newline at end of file
+        other = self.__class__.parse(other)
+        return self >= other
+
+
+def require_gdal_version(version, param=None, values=None, is_max_version=False,
```

Comment:
If the value referenced by `param` is given as a positional argument this decorator will have no idea the value it needs validate has been given.

Reply:
Am I too late to point out that we can have keyword-only arguments in Python 3?

Reply:
I thought about that but it isn't valid syntax for Python 2.